supplemental_notebooks/2021-06-28_muddiest_points.Rmd

```diff
@@ -0,0 +1,192 @@
+
+Unfortunately, the full answer is a bit technical, but the basic answer is this: 
+If you are using functions from the `tidyverse`, you _usually_ don't neeed to use quotes for columns names, and can treat them just like variable names.
+The tidyverse functions will assume that something that looks like a variable name may be a column, so it will look in the data frame for that column (or create a column with that name, if that is the task at hand). 
+
+Base R functions don't do this, so you usually have to put the column names in quotes so that R knows they are character (text) strings.
+The main exception is when you are using the `$` notation, which is explicitly telling R to use a column name. 
+
+Let's look at some examples with the `palmerpengins` data.
+
+In this notebook, I am not going to load any packages except `magrittr` for the `%>%` pipe, so you will see lots of use of the `::` operator.
+I am doing this to make it always clear when I am using package functions, but in many cases you might have loaded `dplyr` and can leave the first part off.
+
+First we will load `magrittr` package and the palmer penguins data frame into a variable called `penguins`, as we did before.
+
+```{r}
+library(magrittr)
+penguins <- palmerpenguins::penguins
+```
+
+Now we can look at what the column names are:
+```{r}
+colnames(penguins)
+```
+
+As you can see, R reports these as character strings, enclosed in quotes.
+
+### Selecting columns
+If we want to select an individual column as a **vector**, we can use the `$`:
+
+```{r}
+bill_lengths <- penguins$bill_length_mm
+
+# look at the structure of the vector we extracted (so we don't print it all)
+str(bill_lengths)
+```
+
+Alternative, we could use a `tidyverse` function from the `dplyr` package called `pull()` to accomplish the same task.
+This function is particularly useful when we need to extract a column in a series of piped (`%>%`) steps, where the `$` would be hard to use.
+Because it is a tidyverse function, no quotes are needed!
+
+```{r}
+bill_depths <- penguins %>%
+  dplyr::pull(bill_depth_mm)
+
+str(bill_depths)
+```
+
+If we want to extract multiple columns into a new data frame, we can do that with the bracket `[]` notation in base R.
+For this we need to have the column names in quotes, and as part of a vector.
+Remember that columns are the second axis, so we will have a blank space before the comma to indicate we want all rows.
+
+```{r}
+penguin_sites <- penguins[, c("species", "island", "year")]
+head(penguin_sites)
+```
+The alternative with `tidyverse` uses `dplyr::select()` and no quotes (no vector either!).
+In this case I am not saving the results to a new variable, just printing the same part as above.
+
+```{r}
+penguins %>% 
+  dplyr::select(species, island, year) %>%
+  head()
+```
+### Filtering
+
+We didn't cover it in the lecture, but it is possible to do filtering by rows in base R too.
+First let's look at a tidyverse example:
```

Comment:
Can we maybe add sentence or comment here telling folks what we're filtering by so they know what to look out for in the results of the below chunk?

Reply:
Done!